cni/network/invoker_cns.go

```diff
@@ -229,20 +280,42 @@ func (invoker *CNSIPAMInvoker) Delete(address *net.IPNet, nwCfg *cni.NetworkConf
 		return errEmptyCNIArgs
 	}
 
-	req := cns.IPConfigRequest{
+	req := cns.IPConfigsRequest{
 		OrchestratorContext: orchestratorContext,
 		PodInterfaceID:      GetEndpointID(args),
 		InfraContainerID:    args.ContainerID,
 	}
 
-	if address != nil {
-		req.DesiredIPAddress = address.IP.String()
+	if len(addresses) > 0 {
+		req.DesiredIPAddresses = make([]string, len(addresses))
+		for i, ipAddress := range addresses {
+			req.DesiredIPAddresses[i] = ipAddress.IP.String()
+		}
 	} else {
 		log.Printf("CNS invoker called with empty IP address")
 	}
 
-	if err := invoker.cnsClient.ReleaseIPAddress(context.TODO(), req); err != nil {
-		return errors.Wrap(err, fmt.Sprintf("failed to release IP %v with err ", address)+"%w")
+	if err := invoker.cnsClient.ReleaseIPs(context.TODO(), req); err != nil {
+		// if we fail a release with a 404 error try using the old API
+		if errors.Is(err, cnscli.ErrAPINotFound) {
+			ipconfigRequest := cns.IPConfigRequest{
+				OrchestratorContext: orchestratorContext,
+				PodInterfaceID:      GetEndpointID(args),
+				InfraContainerID:    args.ContainerID,
+				DesiredIPAddress:    req.DesiredIPAddresses[0],
```

Comment:
req.DesiredIPAddresses[0] can run into nil pointer deref?
what if len(addresses) check on line 289 fails? You will never initialize the slice on line 290.
Has this been tested with UT?